showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -127,6 +155,93 @@ class ShowkaseProcessor @JvmOverloads constructor(
             }.flatten().mapNotNull { it }.toSet()
     }
 
+    private fun processCustomAnnotation(
+        roundEnvironment: XRoundEnv,
+        annotation: XTypeElement? = null
+    ): Set<ShowkaseMetadata.Component> {
+        val supportedTypes = mutableListOf<String>()
+        if (annotation != null) supportedTypes.add(annotation.qualifiedName)
+        supportedTypes.addAll(supportedCustomAnnotationTypes())
+        val components = mutableSetOf<ShowkaseMetadata.Component>()
+
+        supportedTypes.map { supportedType ->
+            val annotatedElements = roundEnvironment.getElementsAnnotatedWith(supportedType)
+            annotatedElements.map { annotatedElement ->
+                if (showkaseValidator.checkElementIsAnnotationClass(annotatedElement)) {
+                    processCustomAnnotation(roundEnvironment, annotatedElement)
```

Comment:
Why do you need to call this recursively? 🤔 I didn't follow this part

Reply:
Initially I thought that if the element was a preview annotation I would like to call this to get the element that it is annotation. However, when thinking about it, I realise now that that will be picked up by when we check for preview annotations. Updated this in [d40c9d5](https://github.com/airbnb/Showkase/pull/303/commits/d40c9d5294899f92367fe0de61095c5336fecad4) :) Thanks!